pandas/core/_numba/extensions.py

```diff
@@ -0,0 +1,541 @@
+# Disable type checking for this module since numba's internals
+# are not typed, and we use numba's internals via its extension API
+# mypy: ignore-errors
+"""
+Utility classes/functions to let numba recognize
+pandas Index/Series/DataFrame
+
+Mostly vendored from https://github.com/numba/numba/blob/main/numba/tests/pdlike_usecase.py
+"""
+
+from __future__ import annotations
+
+import operator
+
+import numba
+from numba.core import (
+    cgutils,
+    types,
+)
+from numba.core.datamodel import models
+from numba.core.extending import (
+    NativeValue,
+    box,
+    lower_builtin,
+    make_attribute_wrapper,
+    overload,
+    overload_attribute,
+    overload_method,
+    register_model,
+    type_callable,
+    typeof_impl,
+    unbox,
+)
+from numba.core.imputils import impl_ret_borrowed
+import numpy as np
+
+from pandas.core.indexes.base import Index
+from pandas.core.indexing import _iLocIndexer
+from pandas.core.series import Series
+
+
+# TODO: Range index support
+# (not passing an index to series constructor doesn't work)
+class IndexType(types.Type):
```

Comment:
This block just defines the types for Index and Series, there isn't much to see here.

It is pretty boilerplate and standard.